src/main/java/org/graylog/integrations/aws/codecs/AWSCodec.java

```diff
@@ -0,0 +1,80 @@
+package org.graylog.integrations.aws.codecs;
+
+import com.google.inject.assistedinject.Assisted;
+import org.graylog.integrations.aws.AWSMessageType;
+import org.graylog.integrations.aws.inputs.AWSInput;
+import org.graylog2.plugin.Message;
+import org.graylog2.plugin.configuration.Configuration;
+import org.graylog2.plugin.configuration.ConfigurationRequest;
+import org.graylog2.plugin.inputs.annotations.ConfigClass;
+import org.graylog2.plugin.inputs.annotations.FactoryClass;
+import org.graylog2.plugin.inputs.codecs.AbstractCodec;
+import org.graylog2.plugin.inputs.codecs.Codec;
+import org.graylog2.plugin.journal.RawMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.util.Map;
+
+public class AWSCodec extends AbstractCodec {
+
+    public static final String NAME = "AWSMetaCodec";
```

Comment:
How about renaming this to `AWSCodec` too?

```suggestion
    public static final String NAME = "AWSCodec";
```